private/bufpkg/bufplugin/parse.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
```

Comment:
I thought this stuff was being pulled out to a common `bufparse` package? Also would likely result in some of `errors.go` being pulled out (and potentially other stuff).

Reply:
I did look at pulling it out into `bufparse` but reverted in 477557f. I noticed we already had another `ParseError` type in `bufcas` so thought copying was more consistent. Will refactor back out to a package.